src/main/java/synapseawsconsolelogin/Auth.java

```diff
@@ -390,28 +593,26 @@ private void doGetIntern(HttpServletRequest req, HttpServletResponse resp)
 				resp.setStatus(200);
 				return;
 			}
-
-			AWSSecurityTokenService stsClient = AWSSecurityTokenServiceClientBuilder.standard()
-					.withRegion(Regions.fromName(awsRegion)).build();
-			
-			AssumeRoleRequest assumeRoleRequest = createAssumeRoleRequest(claims, roleArn, selectedTeam);
-			
-			AssumeRoleResult assumeRoleResult = stsClient.assumeRole(assumeRoleRequest);
-			Credentials credentials = assumeRoleResult.getCredentials();
-			// redirect to AWS login
-			String redirectURL = getConsoleLoginURL(req, credentials, new HttpGetExecutor() {
-
-				@Override
-				public String executeHttpGet(String urlString) throws IOException {
-					URL url = new URL(urlString);
-					URLConnection conn = url.openConnection();
-					BufferedReader bufferReader = new BufferedReader(
-							new InputStreamReader(conn.getInputStream()));  
-					return bufferReader.readLine();
-				}});
 			
-			resp.setHeader(LOCATION, redirectURL);
-			resp.setStatus(303);
+			// we take different actions for different services
+			switch(RequestType.valueOf(state)) {
+			case SC_CONSOLE:
+				redirectToSCConsole(claims, roleArn, selectedTeam, HTTP_EXECUTOR, req, resp);
+				break;
+			case STS_TOKEN:
+				returnStsToken(claims, roleArn, selectedTeam, resp);
+				break;
+			case ID_TOKEN:
+				// creates file for use here: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-role.html#cli-configure-role-oidc
+				returnOidcToken(tokens.getIdToken().getToken(), resp);
+				break;
+			case ACCESS_TOKEN:
+				returnOidcToken(tokens.getAccessToken().getToken(), resp);
+				break;
```

Comment:
can we return this info as json to the user instead of in a file? That will give the user the option to do whatever they want with the data. They may want to easily parse the data, set it to env vars, pipe to file or directly pass it along to another app.

Reply:
What's would the user type at the command line to pass JSON to `aws ssm start-session`?

Reply:
yes, It looks like aws has a way to support sourcing credentials from external process.  https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sourcing-external.html